doc/OnlineDocs/contributed_packages/doe/doe.rst

```diff
@@ -0,0 +1,421 @@
+    ...     # === Define variables ===
+    ...     m.CA0 = pyo.Var(m.t0, initialize = CA_init, bounds=(1.0,5.0), within=pyo.NonNegativeReals) # mol/L
+    ...     m.T = pyo.Var(m.t, initialize =T_initial, bounds=(300, 700), within=pyo.NonNegativeReals)
+    ...     m.C = pyo.Var(m.scena, m.y_set, m.t, initialize=3, within=pyo.NonNegativeReals)
+    ...     m.dCdt = DerivativeVar(m.C, wrt=m.t)
+    ...     m.kp1 = pyo.Var(m.scena, m.t, initialize=3)
+    ...     m.kp2 = pyo.Var(m.scena, m.t, initialize=1)
+    ...     # === Define Param ===
+    ...     m.A1 = pyo.Param(m.scena, initialize=scena['A1'],mutable=True)
+    ...     m.A2 = pyo.Param(m.scena, initialize=scena['A2'],mutable=True)
+    ...     m.E1 = pyo.Param(m.scena, initialize=scena['E1'],mutable=True)
+    ...     m.E2 = pyo.Param(m.scena, initialize=scena['E2'],mutable=True)
+    ...     # === Constraints ===
+    ...     def T_control(m,t):
+    ...         if t in m.t_con:
+    ...             return pyo.Constraint.Skip
+    ...         else:
+    ...             j = -1
+    ...             for t_con in m.t_con:
+    ...                 if t>t_con:
+    ...                     j+=1
+    ...             neighbour_t = t_control[j]
+    ...         return m.T[t] == m.T[neighbour_t]
+    ...     def cal_kp1(m,z,t):
+    ...         return m.kp1[z,t] == m.A1[z]*pyo.exp(-m.E1[z]*1000/(m.R*m.T[t]))
+    ...     def cal_kp2(m,z,t):
+    ...         return m.kp2[z,t] == m.A2[z]*pyo.exp(-m.E2[z]*1000/(m.R*m.T[t]))
+    ...     def dCdt_control(m,z,y,t):
+    ...         if y=='CA':
+    ...             return m.dCdt[z,y,t] == -m.kp1[z,t]*m.C[z,'CA',t]
+    ...         elif y=='CB':
+    ...             return m.dCdt[z,y,t] == m.kp1[z,t]*m.C[z,'CA',t] - m.kp2[z,t]*m.C[z,'CB',t]
+    ...         elif y=='CC':
+    ...             return pyo.Constraint.Skip
+    ...     def alge(m,z,t):
+    ...         return m.C[z,'CA',t] + m.C[z,'CB',t] + m.C[z,'CC', t] == m.CA0[0]
+    ...     m.T_rule = pyo.Constraint(m.t, rule=T_control)
+    ...     m.k1_pert_rule = pyo.Constraint(m.scena, m.t, rule=cal_kp1)
+    ...     m.k2_pert_rule = pyo.Constraint(m.scena, m.t, rule=cal_kp2)
+    ...     m.dCdt_rule = pyo.Constraint(m.scena,m.y_set, m.t, rule=dCdt_control)
+    ...     m.alge_rule = pyo.Constraint(m.scena, m.t, rule=alge)
+    ...     for z in m.scena:
+    ...         m.C[z,'CB',0.0].fix(0.0)
+    ...         m.C[z,'CC',0.0].fix(0.0)
+    ...     return m
+.. doctest::
+
+    >>> # === Discretization ===
+    >>> def disc(m, NFE=32):
+    ...     discretizer = pyo.TransformationFactory('dae.collocation')
+    ...     discretizer.apply_to(m, nfe=NFE, ncp=3, wrt=m.t)
+    ...     return m
+
+.. note::
+    The first argument of the ``create_model`` function should be ``scena``.
+
+.. note::
+    The model parameters (  :math:`A_1, A_2, E_1, E_2`) are defined as either ``Var`` or ``Param`` objects. This is suggested since it allows an easy transition when changing mode to ``direct_kaug``.
```

Comment:
This is because the direct_kaug mode requires parameters to be defined as ``var``, while other modes can deal with parameters as either ``var`` or ``Param`` objects. I change the line 72 so that the discriptions are consistent.